assets/js/components/settings/SettingsModules.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * SettingsModules component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Redirect, Route, Switch, useHistory } from 'react-router-dom';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { getModulesData } from '../../util';
+import SettingsAdmin from './SettingsAdmin';
+import SettingsActiveModules from './SettingsActiveModules';
+import SettingsInactiveModules from './SettingsInactiveModules';
+
+function SettingsModules( props ) {
+	const { _hideExcept } = props;
+	const modulesData = getModulesData();
+
+	const history = useHistory();
+
+	useEffect( () => {
+		if ( global._googlesitekitLegacyData.editmodule && modulesData[ global._googlesitekitLegacyData.editmodule ].active ) {
+			history.push( `/connected-services/${ global._googlesitekitLegacyData.editmodule }/edit` );
+		}
+	}, [ history, modulesData ] );
+
+	if ( ! Object.values( modulesData ).length ) {
+		return null;
+	}
+
+	return (
+		<Switch>
+			{ /* Settings Module Routes */ }
+			<Route path="/connected-services/:moduleSlug/:action">
+				<SettingsActiveModules _hideExcept={ _hideExcept } />
+			</Route>
+			<Route path="/connected-services/:moduleSlug">
+				<SettingsActiveModules _hideExcept={ _hideExcept } />
+			</Route>
+			<Route path="/connected-services">
+				<SettingsActiveModules _hideExcept={ _hideExcept } />
```

Comment:
The `_hideExcept` prop is a bit confusing and seems to only be used in Storybook. Can you elaborate on this?

Reply:
It was kinda a hack, and I figured out a _slightly_ better approach in d778e6b404670e446c394e925c6e7a77e0caac94, let me know what you think 🙂

Reply:
I much prefer the weirdness to live in Storybook than using the previous prop 👍